src/cpp/rtps/transport/TCPv4Transport.cpp

```diff
@@ -76,9 +76,27 @@ TCPv4Transport::TCPv4Transport(
     : TCPTransportInterface(LOCATOR_KIND_TCPv4)
     , configuration_(descriptor)
 {
-    for (const auto& interface : descriptor.interfaceWhiteList)
+    if (!descriptor.interfaceWhiteList.empty())
     {
-        interface_whitelist_.emplace_back(ip::address_v4::from_string(interface));
+        const auto white_begin = descriptor.interfaceWhiteList.begin();
+        const auto white_end = descriptor.interfaceWhiteList.end();
+
+        std::vector<IPFinder::info_IP> local_interfaces;
+        get_ipv4s(local_interfaces, true);
+        for (const IPFinder::info_IP& infoIP : local_interfaces)
+        {
+            if (std::find(white_begin, white_end,
+                    infoIP.name) != white_end ||
+                    std::find(white_begin, white_end, infoIP.dev) != white_end)
+            {
+                interface_whitelist_.emplace_back(ip::address_v4::from_string(infoIP.name));
+            }
+        }
+
+        if (interface_whitelist_.empty())
+        {
+            EPROSIMA_LOG_ERROR(TRANSPORT, "All whitelist interfaces were filtered out");
```

Comment:
Note that without pushing a dummy ip (as it is done in UDP), this would still "work" (i.e. same behavior as not including a whitelist). Perhaps we should also add it here to force the user fix its mistaken configuration.